examples/soap-calculator/test/acceptance/application.acceptance.ts

```diff
@@ -0,0 +1,80 @@
+import {createClientForHandler, supertest} from '@loopback/testlab';
+import {RestServer} from '@loopback/rest';
+import {SoapCalculatorApplication} from '../../src/application';
+import {expect} from '@loopback/testlab';
+import {givenAConnectedDataSource} from '../helpers';
+
+describe('Application', function() {
+  let app: SoapCalculatorApplication;
+  let server: RestServer;
+  let client: supertest.SuperTest<supertest.Test>;
+
+  // tslint:disable-next-line:no-invalid-this
+  this.timeout(5000);
+
+  before(givenAnApplication);
+  before(givenARestServer);
+  before(async () => {
+    await givenAConnectedDataSource();
+  });
+
+  before(async () => {
+    await app.boot();
+    await app.start();
+  });
+
+  before(() => {
+    client = createClientForHandler(server.requestHandler);
+  });
+
+  after(async () => {
+    await app.stop();
+  });
+
+  it('should fail with 412 error', async () => {
+    await client.get('/divide/20/0').expect(412);
+  });
+
+  it('should fail with 404 error', async () => {
```

Comment:
One (logical) assertion per test - please split this test into multiple smaller ones.

```ts
it('returns 404 when the "/divide" is missing required args', async () => {
  await client.get('/divide/').expect(404);
});
```